projects/views.py

```diff
@@ -407,6 +410,27 @@ def post(self, request, *args, **kwargs):
         name = request.POST.get("name", "default")[:200]
         description = request.POST.get("description", "")
 
+        # Ensure no duplicate project name for the common user
+        current_user_previous_project_names = list(Project.objects.filter(owner=request.user).values("name", "status"))
+
+        project_name_already_exists = any(
+            current_user_project["name"] == name and current_user_project["status"] != "deleted"
+            for current_user_project in current_user_previous_project_names
+        )
```

Comment:
```suggestion
        # Ensure no duplicate project name for the common user
        project_name_already_exists = Project.objects.filter(
            owner=request.user,
            name=name,
        ).exclude(status="deleted").exists()
```

Reply:
Thanks. changed the code according to the suggestion.